chronos/asyncmacro2.nim

```diff
@@ -70,21 +72,23 @@ else:
         # because you're calling a callback or forward declaration in your code
         # for which the compiler cannot deduce raises signatures - make sure
         # to annotate both forward declarations and `proc` types with `raises`!
-        if not(nameIterVar.finished()):
-          var next = nameIterVar()
-          # Continue while the yielded future is already finished.
-          while (not next.isNil()) and next.finished():
-            next = nameIterVar()
-            if nameIterVar.finished():
-              break
-
-          if next == nil:
-            if not(retFutureSym.finished()):
-              const msg = "Async procedure (&" & strName & ") yielded `nil`, " &
-                          "are you await'ing a `nil` Future?"
-              raiseAssert msg
-          else:
-            next.addCallback(identName)
+        {.gcsafe.}:
```

Comment:
This problem should be fixed in Nim, we should not hide/workaround problem in our code.